.github/workflows/system.yaml

```
@@ -0,0 +1,35 @@
+name: System tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  system_tests:
+    runs-on: ubuntu-latest
```

Note that in GitHub Action, Ubuntu latest is not yet ... 20.04 ... but I think that is fine for pydoctor and eventually `ubuntu-latest` will get there,

Note that you can also have all your tests into a single workflow and set a custom name for each job.
The test will still be run in parallel.

```suggestion
    name: name: System tests
    runs-on: ubuntu-latest
```

The main advantage is that you have all the test related info for GitHub Action into a single file.

The disadvantage is that if a job fail, when you retry it , it will run all the others from the same workflow.

---

> Note that in GitHub Action, Ubuntu latest is not yet ... 20.04 ... but I think that is fine for pydoctor and eventually `ubuntu-latest` will get there,

I wasn't sure whether Python 3.9 was available on 18.04, so for the unit tests I stuck to the version Twisted uses, which was version 20.04.

> Note that you can also have all your tests into a single workflow and set a custom name for each job.
> The test will still be run in parallel.
> 
> The main advantage is that you have all the test related info for GitHub Action into a single file.

How much duplication could be removed by merging the files? If it's only the `on:` block, it doesn't help that much, but if the setup, install tox and system info logging steps could be included only once it would simplify things. I wasn't sure that would be an option though, because the unit test should use a matrix while doing that for the system test and static code checks would be overkill.

> The disadvantage is that if a job fail, when you retry it , it will run all the others from the same workflow.

pydoctor currently doesn't have any flaky tests and I'd like to keep it that way ;) But the CI environment or services we contact can also be flaky, so we might sometimes need reruns nevertheless.